backend/src/apiserver/list/list_request_util.go

```diff
@@ -0,0 +1,71 @@
+package list
+
+import (
+	"net/url"
+
+	"github.com/golang/protobuf/jsonpb"
+
+	api "github.com/kubeflow/pipelines/backend/api/go_client"
+	"github.com/kubeflow/pipelines/backend/src/common/util"
+)
+
+// parseAPIFilter attempts to decode a url-encoded JSON-stringified api
+// filter object. An empty string is considered valid input, and equivalent to
+// the nil filter, which trivially does nothing.
+func parseAPIFilter(encoded string) (*api.Filter, error) {
```

Comment:
No, I don't. 
Now,  `parseAPIFilter`  being part of `list` package is used by both v1 and v2 via  [ValidatedListOptions](https://github.com/difince/pipelines/blob/a111784eff679ffc76d57f7d2920e1ee0794dfec/backend/src/apiserver/list/list_request_util.go#L36-L43)

Reply:
To add on Anna, we need to carefully handle:
- All POST ResourceManager methods, as they accept as a param v1 specific api Object 
- Need to appropriately handle all defined in *.proto files EMUNs, as they are used everywhere including *_storage.go files.

Reply:
So not only does server package need to be changed according to v2 api, but also resource manager.

Reply:
> How do we want to handle common proto files that do not change between v1 and v2. Do we want to duplicate them or look into separating them out into common / v1 / v2?

It feels like proto files can be separated between v1 and v2, even if some files may be duplicated. They are essentially two sets of APIs. 

> * How do we handle functions like parseAPIFilter that depend on the (common / v1) proto files. In this case, we need to have the same exact function with different return types (*api.Filter vs *apiv2beta1.Filter). One way to go about it is to add an if/else statement or another way is to duplicate the code for v2 (duplicate in the same file or a new file).
>     * Note: while some are simple as if/else, v2 changes the parameter type and removes resource references. Therefore, some functions may require complete re-write

If the function requires a re-write, we will have to write a separate function. As to the ones solvable by a simple if/else, both ways are ok with me, since future removal of v1 won't be of much work either way. 


> To add on Anna, we need to carefully handle:
> * All POST ResourceManager methods, as they accept as a param v1 specific api Object s

Looks like the following POST functions in `resource_manager.go` took in v1 specific api objects, not counting the unexposed functions:
* CreateRun()
* CreateTask()
* CreateJob()
* ReportMetric()
* CreatePipelineVersion()

A quick look at these functions gives me the impression that these v1 api objects are quite deeply ingrained in these functions. We might have to write separate functions for v2, as well. 

> * Need to appropriately handle all defined in *.proto files EMUNs, as they are used everywhere including *_storage.go files.

Did you mean ENUMs, or am I missing something? Could you please show me an example where *.proto are used in *_storage.go files? As far as I can see, all input params are all model objects, which are not affected by *.proto. I do see several constants defined in api being used here ([example](https://github.com/difince/pipelines/blob/127dab4d4671849d596c455f4619ea807d22f6ea/backend/src/apiserver/storage/run_store.go#L355)), but as they are not lots of code, maybe some if/else branch will be sufficient?


These are just my preliminary thoughts. WDYT?

Reply:
> Did you mean ENUMs, or am I missing something? Could you please show me an example where *.proto are used in *_storage.go files? As far as I can see, all input params are all model objects, which are not affected by *.proto. I do see several constants defined in api being used here ([example](https://github.com/difince/pipelines/blob/127dab4d4671849d596c455f4619ea807d22f6ea/backend/src/apiserver/storage/run_store.go#L355)), but as they are not lots of code, maybe some if/else branch will be sufficient?

yes, I meant ENUMS. The example you provide comes from an ENUM which is defined [here](https://github.com/difince/pipelines/blob/5b5f320736064a62daf4f527f58512c8ec5b1d32/backend/api/v2beta1/run.proto#L235-L238) . In v2 we do not change the ENUMs. What could be the condition of the if clause .. ?  And it would be strange having "if"  and "else" that are actually equivalent...

Reply:
> yes, I meant ENUMS. The example you provide comes from an ENUM which is defined [here](https://github.com/difince/pipelines/blob/5b5f320736064a62daf4f527f58512c8ec5b1d32/backend/api/v2beta1/run.proto#L235-L238) . In v2 we do not change the ENUMs. What could be the condition of the if clause .. ? And it would be strange having "if" and "else" that are actually equivalent...

Ah I see! In the example that I linked above, the line `else if r.StorageState != api.Run_STORAGESTATE_AVAILABLE.String()` has to be altered so `r.StorageState` can be compared against both `apiv1beta1.Run_STORAGESTATE_AVAILABLE.String()` and `apiv1beta2.Run_STORAGESTATE_AVAILABLE.String()` and decide whether v1 or v2 it came from. That's what I meant.

By the way, as a non-goal for now, maybe we can consider decoupling `resource_manage` from APIs. This means we convert the API objects into model objects on the server level.